<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>752</width>
    <height>490</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>nml c++ example</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(51, 57, 59);
color: rgb(255, 255, 255);
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>9</y>
      <width>601</width>
      <height>458</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="9" column="1">
      <widget class="QPushButton" name="pushButton_jog_y_plus">
       <property name="text">
        <string>y+</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QPushButton" name="pushButton_free">
       <property name="text">
        <string>free</string>
       </property>
      </widget>
     </item>
     <item row="9" column="0">
      <widget class="QPushButton" name="pushButton_jog_y_min">
       <property name="text">
        <string>y-</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QPushButton" name="pushButton_auto">
       <property name="text">
        <string>auto</string>
       </property>
      </widget>
     </item>
     <item row="15" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>S code :</string>
       </property>
      </widget>
     </item>
     <item row="3" column="2">
      <widget class="QPushButton" name="pushButton_home_z">
       <property name="text">
        <string>home z</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QPushButton" name="pushButton_spindle_on_cw">
       <property name="text">
        <string>spinde on cw</string>
       </property>
      </widget>
     </item>
     <item row="5" column="2">
      <widget class="QPushButton" name="pushButton_spindle_off">
       <property name="text">
        <string>spindle off</string>
       </property>
      </widget>
     </item>
     <item row="6" column="2">
      <widget class="QPushButton" name="pushButton_resume">
       <property name="text">
        <string>resume</string>
       </property>
      </widget>
     </item>
     <item row="15" column="1" colspan="2">
      <widget class="QLabel" name="label_active_scodes">
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
     <item row="6" column="3">
      <widget class="QPushButton" name="pushButton_reverse">
       <property name="text">
        <string>reverse</string>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <widget class="QPushButton" name="pushButton_jog_x_plus">
       <property name="text">
        <string> x+</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QPushButton" name="pushButton_estop">
       <property name="styleSheet">
        <string notr="true">background-color: rgb(51, 57, 59);
color: rgb(255, 255, 255);
</string>
       </property>
       <property name="text">
        <string>e-stop</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="pushButton_machine_on">
       <property name="text">
        <string>machine on</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="10" column="0">
      <widget class="QPushButton" name="pushButton_jog_z_min">
       <property name="text">
        <string>z-</string>
       </property>
      </widget>
     </item>
     <item row="13" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>M code :</string>
       </property>
      </widget>
     </item>
     <item row="12" column="1" colspan="4">
      <widget class="QLabel" name="label_active_gcodes">
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
     <item row="12" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>G code :</string>
       </property>
      </widget>
     </item>
     <item row="8" column="0">
      <widget class="QPushButton" name="pushButton_jog_x_min">
       <property name="text">
        <string>x-</string>
       </property>
      </widget>
     </item>
     <item row="7" column="0">
      <widget class="QPushButton" name="pushButton_load">
       <property name="text">
        <string>load</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QPushButton" name="pushButton_coord">
       <property name="text">
        <string>coord</string>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QPushButton" name="pushButton_run">
       <property name="text">
        <string>run</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QPushButton" name="pushButton_home_y">
       <property name="text">
        <string>home y</string>
       </property>
      </widget>
     </item>
     <item row="13" column="1" colspan="2">
      <widget class="QLabel" name="label_active_mcodes">
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
     <item row="16" column="0">
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="3" column="0">
      <widget class="QPushButton" name="pushButton_home_x">
       <property name="text">
        <string>home x</string>
       </property>
      </widget>
     </item>
     <item row="4" column="2">
      <widget class="QPushButton" name="pushButton_manual">
       <property name="text">
        <string>manual</string>
       </property>
      </widget>
     </item>
     <item row="14" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>F code :</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QPushButton" name="pushButton_spindle_on_ccw">
       <property name="text">
        <string>spindle on ccw</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QPushButton" name="pushButton_mdi">
       <property name="text">
        <string>mdi</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QPushButton" name="pushButton_teleop">
       <property name="text">
        <string>teleop</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QPushButton" name="pushButton_pause">
       <property name="text">
        <string>pause</string>
       </property>
      </widget>
     </item>
     <item row="14" column="1" colspan="2">
      <widget class="QLabel" name="label_active_fcodes">
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
     <item row="10" column="1">
      <widget class="QPushButton" name="pushButton_jog_z_plus">
       <property name="text">
        <string>z+</string>
       </property>
      </widget>
     </item>
     <item row="6" column="4">
      <widget class="QPushButton" name="pushButton_stop">
       <property name="text">
        <string>stop</string>
       </property>
      </widget>
     </item>
     <item row="11" column="0">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>jog speed</string>
       </property>
      </widget>
     </item>
     <item row="11" column="2" colspan="2">
      <widget class="QScrollBar" name="horizontalScrollBar_jog_speed">
       <property name="maximum">
        <number>2000</number>
       </property>
       <property name="value">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="11" column="1">
      <widget class="QLabel" name="label_jog_speed">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="5" column="3">
      <widget class="QLabel" name="label_mode">
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
     <item row="3" column="3">
      <widget class="QPushButton" name="pushButton_home_all">
       <property name="text">
        <string>home all</string>
       </property>
      </widget>
     </item>
     <item row="3" column="4">
      <widget class="QPushButton" name="pushButton_unhome_all">
       <property name="text">
        <string>unhome all</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>752</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
